src/DotNet/Writer/Metadata.cs

```diff
@@ -1608,16 +1621,18 @@ internal sealed class Rows<T> where T : class {
 					AddFieldRVA(field);
 					AddImplMap(new MDToken(Table.Field, rid), field);
 					AddConstant(new MDToken(Table.Field, rid), field);
+					errorContext.Source = null;
```

Comment:
This code should initialize `Source` to the previous value which was initialized above to`errorContext.Source = type;`. There's other code in this file that also overwrites it with null when the old value should be used.

The best solution is to create a new disposable `struct` that initializes `Source` and saves the original value and restores it in `Dispose()`.